### PR TITLE
Metrics mpi

### PR DESCRIPTION
**Pull request by rouga**
+ Integration of connectivity with MPI
+ Integration of Metrics with MPI
+ Unit tests for Metrics

### CODE REVIEW

File: include/utils/Vector3.h [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+		static Vector3<double> crossProduct(Vector3& A, Vector3& B){
+		        return Vector3(A.y * B.z - A.z * B.y,
+		                       A.z * B.x - A.x * B.z,
+		                       A.x * B.y - A.y * B.x);
+		}
+
+
+		Vector3
+		operator+(double s) {
```
Comment (abraham-bconstant) [inline include/utils/Vector3.h:81]: Should the argument type be T? This formulation might cause issues with an int vector3?

File: include/mesh/Metrics.hpp [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+		void computeFaceMetrics();
+
+        /**
+         * @brief Compute cell metrics by populating member variables _CellVolumes and _CellCentroids
+         */
+		void computeCellMetrics();
+
+
+		const Mesh<Parser::MeshPartition> &_localMesh;
+		std::vector<Vector3<double>> _faceNormals;
```
Comment (abraham-bconstant) [inline include/mesh/Metrics.hpp:40]: I do not think the data will be contiguous. Something to keep in mind if we have performance issues later down the line

File: include/utils/Vector3.h (hunk 1 above)
Comment (rouga) [reply include/utils/Vector3.h:81]: Sure, we can do this for lines 73 and 69 too.

File: include/mesh/Metrics.hpp (hunk 2 above)
Comment (rouga) [reply include/mesh/Metrics.hpp:40]: > I do not think the data will be contiguous. Something to keep in mind if we have performance issues later down the line

_faceNormal isn't contiguous ?

**Pull request merged**